overlord/ifacestate/helpers.go

```diff
@@ -252,11 +252,15 @@ var removeStaleConnections = func(st *state.State) error {
 		return err
 	}
 	var staleConns []string
-	for id := range conns {
+	for id, conn := range conns {
 		connRef, err := interfaces.ParseConnRef(id)
 		if err != nil {
 			return err
 		}
+		// do not remove connections belonging to hotplug devices as they may re-appear
+		if conn.HotplugKey != "" {
+			continue
```

Comment:
do we have a test for this?  is this code even triggered for new stuff or it's here only to repair old things? it would sound the latter from the comment on the function itself?  if this is still needed shouldn't we remove the connection anyway if the plug snap itself is no more?

Reply:
You're right, this is wrong, hotplug shouldn't be special-cased here. Removed.